keda/templates/metrics-server/servicemonitor.yaml

```diff
@@ -29,7 +29,7 @@ spec:
   {{- toYaml . | nindent 4 }}
   {{- end }}
   endpoints:
-  - port: {{ .Values.prometheus.metricServer.portName }}
+  - port: {{ .Values.prometheus.metricServer.serviceMonitor.port }}
```

Comment:
Is this a breaking change? We should avoid them

Reply:
It will be if users are overwriting the `portName` to be different than `serviceMonitor.port` otherwise, it should be fine since both ports need to be the same, and changing to `serviceMonitor.port` follows the same configuration from `servicemonitor.yaml` I can add a check to make sure the port names are the same as well

Reply:
> It will be if users are overwriting the portName to be different than serviceMonitor.port

so, it's a breaking change indeed xD We need to ensure backward compatibility

Reply:
I removed the changes